src/ProductConstructionService/ProductConstructionService.DependencyFlow/PullRequestUpdater.cs

```diff
@@ -149,6 +155,18 @@ public async Task<bool> ProcessPendingUpdatesAsync(SubscriptionUpdateWorkItem up
         }
         else
         {
+            if (checkNextCommitToProcess)
+            {
+                if (!string.IsNullOrEmpty(pr.NextCommitToProcess) && pr.NextCommitToProcess != update.SourceSha)
+                {
+                    _logger.LogInformation("Skipping update for subscription {subscriptionId} commit {updateCommit} since there's a newer commit {nextCommit}",
+                        update.SubscriptionId,
+                        update.SourceSha,
+                        pr.NextCommitToProcess);
+                    return false;
```

Comment:
Do not return false if you don't fail to process the work item. We will have alerting based on failed work items. We processed everything fine here.